### PR TITLE
quincy: MDSAuthCaps: print better error message for perm flag in MDS caps

### PR DESCRIPTION
**Pull request by rishabh-d-dave**
backport tracker: https://tracker.ceph.com/issues/63407

---

backport of https://github.com/ceph/ceph/pull/52042
parent tracker: https://tracker.ceph.com/issues/61666

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by rishabh-d-dave** [comment]
https://pulpito.ceph.com/rishabh-2023-12-19_05:13:44-fs-rishabh-mdscap-quincy-testing-default-smithi
    https://pulpito.ceph.com/rishabh-2023-12-19_12:25:28-fs-rishabh-mdscap-quincy-testing-default-smithi
    https://pulpito.ceph.com/rishabh-2023-12-22_08:14:05-fs-rishabh-mdscap-quincy-testing-default-smithi

**Comment by rishabh-d-dave** [comment]
No cause for failure was found, unlikely that this PR will be related. https://jenkins.ceph.com/job/ceph-api/66223/

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by NitzanMordhai** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/QUINCY#httpstrellocomcq0ZDB0OD1960-wip-yuri5-testing-2024-02-16-1120-quincy

**Comment by rishabh-d-dave** [comment]
Majority of job failures don't look related but few are. Following are a couple of them - job #7567582 and job #7573791.

Both these failures needs PR #52709 to be backported along with this PR.

**Comment by vshankar** [comment]
@rishabh-d-dave please fix jenkins failures and update.

**Comment by rishabh-d-dave** [comment]
All commits here are signed AFAIS -

```
$ git log 1f2f55109dc1ed2dc2dda9de1db3c70923ba0dfd..HEAD | grep 'Signed-off-by'
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
    Signed-off-by: Rishabh Dave <ridave@redhat.com>
$ git log 1f2f55109dc1ed2dc2dda9de1db3c70923ba0dfd..HEAD | grep 'Signed-off-by' | wc -l
15
```

**Comment by vshankar** [comment]
> All commits here are signed AFAIS -

Could you rebase and repush then?

**Comment by rishabh-d-dave** [comment]
> > All commits here are signed AFAIS -
> 
> Could you rebase and repush then?

Done.

File: qa/tasks/ceph_test_case.py [hunk 1]
```diff
@@ -36,9 +136,23 @@ class CephTestCase(unittest.TestCase):
     # their special needs.  If not met, tests will be skipped.
     REQUIRE_MEMSTORE = False
 
+    def _init_mon_manager(self):
+        # if vstart_runner.py has invoked this code
+        if 'Local' in str(type(self.ceph_cluster)):
+            from tasks.vstart_runner import LocalCephManager
+            self.mon_manager = LocalCephManager(ctx=self.ctx)
```
Comment (vshankar) [inline qa/tasks/ceph_test_case.py:143]: @rishabh-d-dave 

```
2024-03-13 19:29:15,638.638 INFO:__main__:----------------------------------------------------------------------
2024-03-13 19:29:15,639.639 INFO:__main__:Traceback (most recent call last):
2024-03-13 19:29:15,639.639 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/mgr/test_dashboard.py", line 16, in setUp
2024-03-13 19:29:15,639.639 INFO:__main__:    super(TestDashboard, self).setUp()
2024-03-13 19:29:15,639.639 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/ceph_test_case.py", line 153, in setUp
2024-03-13 19:29:15,639.639 INFO:__main__:    self._init_mon_manager()
2024-03-13 19:29:15,639.639 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/ceph_test_case.py", line 143, in _init_mon_manager
2024-03-13 19:29:15,639.639 INFO:__main__:    self.mon_manager = LocalCephManager(ctx=self.ctx)
2024-03-13 19:29:15,639.639 INFO:__main__:TypeError: LocalCephManager.__init__() got an unexpected keyword argument 'ctx'
```

**Review by vshankar — CHANGES_REQUESTED** [review]
PTAL at the failure.

File: qa/tasks/ceph_test_case.py (hunk 1 above)
Comment (rishabh-d-dave) [reply qa/tasks/ceph_test_case.py:143]: Recent push should fix it, I'll keep an eye.

**Comment by rishabh-d-dave** [comment]
jenkins test make check

File: qa/tasks/ceph_test_case.py (hunk 1 above)
Comment (rishabh-d-dave) [reply qa/tasks/ceph_test_case.py:143]: Fix worked, API tests passed this time - https://jenkins.ceph.com/job/ceph-api/70575/

**Comment by rishabh-d-dave** [comment]
jenkins test api

**Comment by rishabh-d-dave** [comment]
Last `ceph API tests` failure was unrelated - https://jenkins.ceph.com/job/ceph-api/72209/

**Review by rishabh-d-dave — COMMENTED** [review]
QA run was successful - https://tracker.ceph.com/projects/cephfs/wiki/Quincy#12-April-2024

**Review by mchangir — APPROVED** [review]
LGTM

**Pull request merged**